homework/index.js

```diff
@@ -29,26 +34,128 @@
-      if (err) {
+  function renderContDetails(contributors, contUl) {
+    createAndAppend('h5', contUl, {
+      text: 'Contributions',
+      class: 'cont-title',
+    });
+    return contributors.forEach(contributor => {
+      const li = createAndAppend('li', contUl, { class: 'contribution' });
+      createAndAppend('img', li, {
+        src: contributor.avatar_url,
+        alt: `${contributor.login}'s image`,
+        class: 'img',
+      });
+      createAndAppend('a', li, {
+        href: contributor.html_url,
+        text: contributor.login,
+        target: '_blank',
+        class: 'link',
+      });
+      createAndAppend('span', li, {
+        text: contributor.contributions,
+        class: 'span-elm',
+      });
+    });
+  }
+  function renderContributors(selectedRepo, contUl) {
+    fetchJSON(selectedRepo.contributors_url)
+      .then(contributors => {
+        contUl.innerHTML = '';
+
+        renderContDetails(contributors, contUl);
+      })
+      .catch((err, root) => {
         createAndAppend('div', root, {
           text: err.message,
           class: 'alert-error',
         });
-        return;
-      }
-      const ul = createAndAppend('ul', root);
-      repos.forEach(repo => renderRepoDetails(repo, ul));
+      });
+  }
+  function renderPage(event, reposUl, contUl) {
+    const url = HYF_REPOS_URL;
+    fetchJSON(url)
+      .then(repos => {
+        reposUl.innerHTML = '';
+        const sortedRepos = repos.sort(sortReposByNameProp);
+        const selectedRepo = sortedRepos[event.target.value || 0];
+        renderRepoDetails(selectedRepo, reposUl);
+        renderContributors(selectedRepo, contUl);
+      })
+      .catch((err, root) => {
+        createAndAppend('div', root, {
+          text: err.message,
+          class: 'alert-error',
+        });
+      });
+  }
+
+  function main(url) {
+    const root = document.getElementById('root');
+    const header = createAndAppend('header', root, { class: 'header' });
+    const mainElm = createAndAppend('main', root, { class: 'main-elm' });
+    createAndAppend('span', header, { text: 'HYF Repositories' });
+    const select = createAndAppend('select', header, { class: 'select' });
+    const reposSection = createAndAppend('section', mainElm, {
+      class: 'repos-section',
+    });
+    const contSection = createAndAppend('section', mainElm, {
+      class: 'cont-section',
+    });
+    const reposUl = createAndAppend('ul', reposSection, {
+      class: 'repos-ul',
+    });
+    const contUl = createAndAppend('ul', contSection, { class: 'cont-ul' });
+    select.addEventListener('change', event =>
+      renderPage(event, reposUl, contUl),
+    );
+
+    fetchJSON(url).then(repos => {
+      repos.sort(sortReposByNameProp).forEach((repo, index) => {
+        createAndAppend('option', select, {
+          value: index,
+          text: repo.name,
+        });
+      });
+      renderPage(event, reposUl, contUl);
```

Comment:
The `event` argument used here is undefined. It produces an ESLint error causing your PR to fail the Travis test. It also causes a problem in your `renderPage` function which you tried to fix by adding `0` like this:

```js
const selectedRepo = sortedRepos[event.target.value || 0];
```

This, of course, is not the way to solve this.

In general, you should not pass events to functions other than event handlers. In this case, you could have passed `event.target.value` in the event handler and simply `0` for rendering the first repo at application startup.

However, as I said, you should only fetch the repositories once, at application startup:

```js
function main(url) {
  ...
  fetchJSON(url).then(repos => {
    repos.sort(sortReposByNameProp).forEach((repo, index) => {
      createAndAppend('option', select, {
        value: index,
        text: repo.name,
      });
    });
    select.addEventListener('change', event =>
      renderPage(repos[event.target.value], reposUl, contUl),
    );
    renderPage(repos[0], reposUl, contUl);
  });
}
```

Then, your `renderPage()` function could be simplified to:

```js
function renderPage(selectedRepo, reposUl, contUl) {
  renderRepoDetails(selectedRepo, reposUl);
  renderContributors(selectedRepo, contUl);
}
```